### PR TITLE
EOS-20955: Motr panic during write operation.

### PR DESCRIPTION
**Pull request by gshipra**
EOS-20955: Motr panic during write operation.
Problem:
There is a situation in which S3 code can send
NULL data to m0_client_calculate_pi. There are checks and
asserts in motr code which fails as of today.
Solution:
Update the checks in motr code and allow
callers to pass NULL buffer vec in m0_client_calculate_pi.

Testing
Code compiled and newly written UT passed.
Tried sending NULL buf vec and it worked.

Signed-off-by: Shipra Gupta <shipra.gupta@seagate.com>

### CODE REVIEW

**Comment by gshipra** [comment]
@sg-shankar @nileshgovande - please review the code.

**Review by huanghua78 — APPROVED** [review]

File: motr/client.h [hunk 1]
```diff
@@ -917,6 +917,8 @@ enum
 
 enum m0_pi_calc_flag {
```
Comment (knrajnambiar76) [inline motr/client.h:918]: consider a case where in user passes flag by ORing
ie flag = M0_PI_CALC_UNIT_ZERO | M0_PI_SKIP_CALC_FINAL
here the value of flag is 3 (enum doesn't have value 3)

File: motr/client.c [hunk 2]
```diff
@@ -847,9 +847,9 @@ M0_INTERNAL int m0_calculate_md5_inc_context(
 
 	M0_PRE(pi != NULL);
 	M0_PRE(curr_context != NULL);
-	M0_PRE(bvec != NULL);
-	M0_PRE(bvec->ov_vec.v_count != NULL);
-	M0_PRE(bvec->ov_buf != NULL);
+	M0_PRE(ergo(bvec != NULL && bvec->ov_vec.v_nr != 0,
```
Comment (knrajnambiar76) [inline motr/client.c:850]: if this means we dont assert when bvec->ov_vec.v_nr is 0 or bvec->ov_vec.v_count is 0 or bvec is 0 -- seems to be fine

File: motr/client.h (hunk 1 above)
Comment (gshipra) [reply motr/client.h:918]: flag = M0_PI_CALC_UNIT_ZERO | M0_PI_SKIP_CALC_FINAL
flag value = 3 or 11 in binary

M0_PI_CALC_UNIT_ZERO - 01
M0_PI_SKIP_CALC_FINAL - 10

11 & 01 - true
11 & 10 - true

We do not need to add another flag for 3.

File: motr/client.c (hunk 2 above)
Comment (gshipra) [reply motr/client.c:850]: This means if bvec is NULL or bvec->ov_vec.v_nr == 0, in that case bvec->ov_vec.v_count can be NULL, bvec->ov_buf can be NULL.

**Comment by madhavemuri** [comment]
@yeshpal-jain-seagate : Is this PR still valid one?

**Comment by r-wambui** [comment]
Closing this as per recommendations from @chandradharraval via Motr Support Teams channel.

**Pull request closed without merging**